Expecto/Expecto.fs

```diff
@@ -1612,10 +1623,10 @@ module Tests =
     | Fail_On_Focused_Tests -> fun o -> { o with failOnFocusedTests = true }
     | Debug -> fun o -> { o with verbosity = LogLevel.Debug }
     | Log_Name name -> fun o -> { o with logName = Some name }
-    | Filter hiera -> fun o -> {o with filter = Test.filter (fun s -> s.StartsWith hiera )}
-    | Filter_Test_List name ->  fun o -> {o with filter = Test.filter (fun s -> s |> getTestList o.splitter |> Array.exists(fun s -> s.Contains name )) }
-    | Filter_Test_Case name ->  fun o -> {o with filter = Test.filter (fun s -> s |> getTestCase o.splitter |> fun s -> s.Contains name )}
-    | Run tests -> fun o -> {o with filter = Test.filter (fun s -> tests |> List.exists ((=) s) )}
+    | Filter hiera -> fun o -> {o with filter = Test.filter (fun s -> s |> List.head |> fun z -> z.StartsWith hiera )}
```

Comment:
I don't really like to have thing that may crash inside this codebase, so please instead handle the distinct cases than use head.

Reply:
Agree, removed.